runtime-modules/blog/src/lib.rs

```diff
@@ -578,36 +637,117 @@ decl_module! {
             Ok(())
         }
 
+        /// Remove reply from storage
+        ///
+        /// <weight>
+        ///
+        /// ## Weight
+        /// `O (1)` doesn't depends on the state or parameters
+        /// - DB:
+        ///    - O(1) doesn't depend on the state or parameters
+        /// # </weight>
+        #[weight = BlogWeightInfo::<T, I>::delete_reply()]
+        pub fn delete_replies(
+            origin,
+            participant_id: ParticipantId<T>,
+            replies: Vec<(PostId, T::ReplyId, bool)>,
+        ) -> DispatchResult {
+            let account_id = Self::ensure_valid_participant(origin, participant_id)?;
+
+            let mut erase_replies = Vec::new();
+            for (post_id, reply_id, hide) in replies {
+                // Ensure post with given id exists
+                let post = Self::ensure_post_exists(post_id)?;
+
+                // Ensure post unlocked, so mutations can be performed
+                Self::ensure_post_unlocked(&post)?;
+
+                // Ensure reply with given id exists
+                let reply = Self::ensure_reply_exists(post_id, reply_id)?;
+
+                // Ensure reply -> owner relation exists if post lifetime hasn't ran out
+                if (frame_system::Module::<T>::block_number() - reply.last_edited) <
+                    T::ReplyLifetime::get()
+                {
+                    Self::ensure_reply_ownership(&reply, &participant_id)?;
+                }
+
+                if !reply.is_owner(&participant_id) {
+                    ensure!(!hide, Error::<T, I>::ReplyOwnershipError);
+                }
+
+                erase_replies.push((post_id, reply_id, reply.cleanup_pay_off, hide));
+            }
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            for (post_id, reply_id, cleanup_pay_off, hide) in erase_replies {
+                Self::pay_off(post_id, cleanup_pay_off, &account_id)?;
```

Comment:
It looks like it's possible to rewrite it as a single call (sum pay-off). Each separate `pay_off` call in the loop could issue an error and the state would be corrupted. However, a single call would leave consistent data even in the case of an error.

Reply:
The problem with that is that the pay offs might be in different post accounts, we could however do first a loop doing the pay off for each deleted reply and then this loop but that would have a performance cost and if it errors in any of the pay off after the first it would still be left in an inconsistent state.

Reply:
Adding to our conversation: it makes sense then to check separately that all post accounts have enough funds to make pay-offs (prior to any state mutation).